source/jormungandr/jormungandr/utils.py

```diff
@@ -563,3 +566,85 @@ def worker():
     flask.globals._request_ctx_stack.push(request_context)
     yield
     flask.globals._request_ctx_stack.pop()
+
+
+def compose(*funs):
+    """
+    compose functions and return a callable object
+
+    example 1:
+    f(x) = x + 1
+    g(x) = 2*x
+
+    compose(f,g) = g(f(x)) = 2 * (x + 1 )
+
+    example 2:
+    f(a list of integer): returns multiples of 3
+    g(a list of integer): returns multiples of 5
+
+    compose(f,g): returns multiples of 3 AND 5
+
+    :param funs:
+    :return: a lambda
+
+    >>> c = compose(lambda x: x+1, lambda x: 2*x)
+    >>> c(42)
+    86
+
+    >>> f = lambda l: (x for x in l if x%3 == 0)
+    >>> g = lambda l: (x for x in l if x%5 == 0)
+    >>> c = compose(f, g)
+    >>> list(c(range(45)))
+    [0, 15, 30]
+    """
+    return lambda obj: functools.reduce(lambda prev, f: f(prev), funs, obj)
+
+class ComposedFilter(object):
+    """
+    Compose several filters with convenient interfaces
+    All filters are evaluated lazily
+
+    >>> F = ComposedFilter()
+    >>> f = F.add_filter(lambda x: x % 2 == 0).add_filter(lambda x: x % 5 == 0).compile()
+    >>> list(f(range(40)))
+    [0, 10, 20, 30]
+    >>> list(f(range(20))) # we can reuse the composed filter
+    [0, 10]
+    >>> f = F.add_filter(lambda x: x % 3 == 0).compile() # we can continue on adding new filter
+    >>> list(f(range(40)))
+    [0, 30]
+    """
+    def __init__(self):
+        self.filters = []
+
+    def add_filter(self, pred):
+        self.filters.append(lambda iterable: (i for i in iterable if pred(i)))
+        return self
+
+    def compile(self):
```

Comment:
I would rather see `def compose_filters` or `apply_filters` to be more consistent.